### ISSUE TITLE
Make the docs and examples build optional

### ISSUE THREAD

**Issue by alainsanguinetti** [accellera-official/systemc#23]
Hello, for embedding this library in a larger project, the docs and examples files generate an unnecessary overhead.
There is currently no possibility to make the configure and build skip those folders so it's not possible to remove them without heavy modification of the Makefiles.

I propose to make the configure and build of those folders optional, so that they are only built on demand by the user and that they can be completely removed from the folder if necessary.

**Comment by maehne** [comment]
I agree that providing this option is sensible. This should also be implemented in the CMake-based build system.
